### PR TITLE
docs: readme

### PR DESCRIPTION
**Pull request by 0xShaito**
# 🤖 Linear

Closes GRT-157

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,6 +154,12 @@ yarn coverage
 
 Configure the `.env` variables.
 
+Import your private keys into Foundry's encrypted keystore:
+
+```bash
+cast wallet import $ARBITRUM_DEPLOYER --interactive
```
Comment (0xJabberwock) [inline README.md:160]: `$ARBITRUM_DEPLOYER` needs to be added to `.env.example` and implemented in `yarn deploy:arbitrum`. Also, let's rename it to `$ARBITRUM_DEPLOYER_NAME`.

File: README.md [hunk 2]
```diff
@@ -1,36 +1,84 @@
...
+
+| Yarn Command            | Description                                                |
+| ----------------------- | ---------------------------------------------------------- |
+| `yarn build`            | Compile all contracts.                                     |
+| `yarn coverage`         | See `forge coverage` report.                               |
+| `yarn deploy:arbitrum`  | Deploy the contracts to Arbitrum mainnet.                  |
+| `yarn test`             | Run all unit and integration tests.                        |
+| `yarn test:unit`        | Run unit tests.                                            |
+| `yarn test:deep`        | Run unit tests with 5000 fuzz runs.                        |
+| `yarn test:integration` | Run integration tests.                                     |
```
Comment (0xJabberwock) [inline README.md:33]: There are two setup sections which are more or less redundant. A similar description for all these commands can be found below.

**Review by 0xJabberwock — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -1,36 +1,84 @@
...
+
+## Design
+
+EBO uses [Prophet](https://github.com/defi-wonderland/prophet-core). A versatile and fully adaptable optimistic oracle solution that allows users to set custom modules to achieve the functionality they need. It works by implementing multiple modules to get the resulting block in other chains for an epoch in the protocol chain.
```
Comment (ashitakah) [inline README.md:37]: do we have a book for prophet instead of use the github link

File: README.md [hunk 4]
```diff
@@ -1,36 +1,84 @@
...
-  <dd>Export your Solidity interfaces and contracts as packages, and publish them to NPM.</dd>
-</dl>
+### Periphery
+
+**EBORequestCreator**
+
+Simplifies request creation and validates requested data. 
+
+> [!WARNING]  
+> Only allows one pair of epoch-chainId request to be active at a time.
```
Comment (ashitakah) [inline README.md:73]: now the warning makes no sense because the function only allows to pass 1 by 1

**Review by ashitakah — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (0xShaito) [reply README.md:73]: I modified it. If you create a request for a chainId + epoch you can't create the same one in a different call

File: README.md [hunk 5]
```diff
@@ -1,36 +1,84 @@
...
 
-  <dt>Deployment scripts</dt>
-  <dd>Sample scripts to deploy contracts on both mainnet and testnet.</dd>
+Using a fork of Prophet's [CallbackModule](https://github.com/defi-wonderland/prophet-modules/blob/dev/solidity/contracts/modules/finality/CallbackModule.sol), EBO can emit events with the finalized data so it can be indexed by a subgraph.
 
-  <dt>Sample Integration, Unit, Property-based fuzzed and symbolic tests</dt>
-  <dd>Example tests showcasing mocking, assertions and configuration for mainnet forking. As well it includes everything needed in order to check code coverage.</dd>
-  <dd>Unit tests are built based on the <a href="https://twitter.com/PaulRBerg/status/1682346315806539776">Branched-Tree Technique</a>, using <a href="https://github.com/alexfertel/bulloak">Bulloak</a>.
-  <dd>Formal verification and property-based fuzzing are achieved with <a href="https://github.com/a16z/halmos">Halmos</a> and <a href="https://github.com/crytic/echidna">Echidna</a> (resp.).
+**AccountingExtension**
```
[the remark this answers is not in the document]
Comment (0xShaito) [reply README.md:62]: Moving to periphery

File: README.md (hunk 2 above)
Comment (0xShaito) [reply README.md:33]: I think it's cool to leave it as a summary easy to see

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (0xJabberwock) [reply README.md:65]: Contracts under `Periphery` were those that couldn't be found in Prophet's base architecture; hence why I proposed the `Extensions` category.

**Review by 0xJabberwock — APPROVED** [review]
Notice that `EBO_flows.png` contains some inexact naming.

**Review by ashitakah — APPROVED** [review]

**Pull request merged**